src/sonic-yang-models/yang-models/sonic-stp.yang

```diff
@@ -0,0 +1,236 @@
+module sonic-stp {
+    namespace "http://github.com/sonic-net/sonic-stp";
+    prefix stp;
+
+    description "SONiC STP YANG model for PVST and MST configurations";
+
+    revision 2024-11-22 {
+        description "Version01 for combined PVST and MST configurations.";
+    }
+
+    container stp {
+        description "STP global and per-instance configurations";
+
+        container global {
+            description "Global STP settings";
+            leaf mode {
+                type enumeration {
+                    enum "pvst" {
+                        description "Per VLAN Spanning Tree mode";
+                    }
+                    enum "mst" {
+                        description "Multiple Spanning Tree mode";
+                    }
+                }
+                description "STP mode (PVST or MST)";
+            }
+            leaf rootguard_timeout {
+                type uint16 {
+                    range "5..600";
+                }
+                default "30";
+                description "Root guard timeout in seconds";
+            }
+            leaf forward_delay {
+                type uint8 {
+                    range "4..30";
+                }
+                default "15";
+                description "Forward delay in seconds";
+            }
+            leaf hello_time {
+                type uint8 {
+                    range "1..10";
+                }
+                default "2";
+                description "Hello time in seconds";
+            }
+            leaf max_age {
+                type uint8 {
+                    range "6..40";
+                }
+                default "20";
+                description "Maximum age time in seconds";
+            }
+            leaf priority {
+                type uint16 {
+                    range "0..61440";
+                }
+                default "32768";
+                description "Bridge priority";
+            }
+            
+            // MST-specific attributes
+            leaf mst_name {
+                when "../mode = 'mst'";
+                type string {
+                    length "1..32";
+                }
+                default "<MAC_ADDRESS>";
+                description "MST region name (default: MAC address of switch)";
+            }
+            leaf mst_revision {
+                when "../mode = 'mst'";
+                type uint16 {
+                    range "0..65535";
+                }
+                default "0";
+                description "MST region revision number";
+            }
+            leaf mst_max_hop {
+                when "../mode = 'mst'";
+                type uint8 {
+                    range "1..40";
+                }
+                default "20";
+                description "Maximum hops in MST region";
+            }
+        }
+
+        list vlan {
```

Comment:
Vlan block should be like this 

 container STP_VLAN {
            list STP_VLAN_LIST {
                key "name";
                sonic-ext:custom-validation ValidateStpMaxInstance;
                leaf name {
                    type string;
                    description
                        "Vlan identifier";
                }

                leaf vlanid {
                    type uint16 {
                        range "1..4095" {
                            error-message "Vlan ID out of range";
                            error-app-tag vlanid-invalid;
                        }
                    }
                    description
                        "Vlan identifier number";
                }

                leaf enabled {
                    type boolean;
                    sonic-ext:custom-validation ValidateStpDisableVlan;
                    mandatory true;
                    description
                        "Spanning tree enabled/disabled on Vlan";
                }

                uses vlanModeAttr;
            }
        }

Reply:
Please review, I have updated